### DISCUSSION TITLE
Enable developers to optimize map loading experience for users with slower connection speed

### DISCUSSION THREAD

**Discussion by hy9be** [maplibre/maplibre#338] [Cross-project feature proposals]
Some applications serve user groups with slower connection speed. We would like to propose several enhancements to enable developers to better take care of these users' experience when waiting for map tiles being rendered:

1. A guideline for adding map tile loading progress indicator to the application.
2. A toggle for fetching and rendering tiles at a higher level of zoom level than the current viewport.
3. A toggle for adding a background grid to the canvas like https://github.com/wipfli/background-grid.

Please evaluate. Other suggestions are welcomed. Thank you.

**Comment by sjg-wdw** [comment]
One way we dealt with this in WhirlyGlobe-Maply was to load in lower res tiles first and then replace them with higher.  That let you get an overview quickly which was then replaced with a more accurate version more slowly.  Worked particularly well with rasters.

**Comment by sjg-wdw** [comment]
We also had this idea of "importance" where we calculated the screen space taken up by a given tile and then loaded what was next based on a sorted list of priorities.  The point being, we could tweak the importance for a given layer and load less of it if it was very expensive.
